File: core/repository/repository_intelligence.py
from __future__ import annotations

from typing import Dict, Any
from core.repository.topology_engine import build_topology
from core.repository.dependency_graph_engine import build_dependency_graph
from core.repository.api_surface_engine import extract_api_surface
from core.repository.import_graph_engine import build_import_graph
from core.repository.package_engine import detect_package_managers
from core.repository.architecture_engine import infer_architecture
from core.repository.repo_classifier import classify_repo


def analyze_repository(text: str, source_url: str = "") -> Dict[str, Any]:
    topology = build_topology(text)
    dependency_graph = build_dependency_graph(text)
    api_surface = extract_api_surface(text)
    import_graph = build_import_graph(text)
    packages = detect_package_managers(text)
    architecture = infer_architecture(topology, import_graph)
    classifier = classify_repo(source_url)
    return {
        "topology": topology,
        "dependency_graph": dependency_graph,
        "api_surface": api_surface,
        "import_graph": import_graph,
        "packages": packages,
        "architecture": architecture,
        "classifier": classifier,
    }

File: src/repository/repositoryIntelligence.ts
/**
 * Converted from Python: core/repository/repository_intelligence.py
 * @generated — WebWeaveX python→javascript library port
 */

import { buildTopology } from "./topologyEngine.js";
import { buildDependencyGraph } from "./dependencyGraphEngine.js";
import { extractApiSurface } from "./apiSurfaceEngine.js";
import { buildImportGraph } from "./importGraphEngine.js";
import { detectPackageManagers } from "./packageEngine.js";
import { inferArchitecture } from "./architectureEngine.js";
import { classifyRepo } from "./repoClassifier.js";

export function analyzeRepository(text: any, source_url: any = ""): any {
  var topology: any = buildTopology(text);
  var dependency_graph: any = buildDependencyGraph(text);
  var api_surface: any = extractApiSurface(text);
  var import_graph: any = buildImportGraph(text);
  var packages: any = detectPackageManagers(text);
  var architecture: any = inferArchitecture(topology, import_graph);
  var classifier: any = classifyRepo(source_url);
  return {"topology": topology, "dependency_graph": dependency_graph, "api_surface": api_surface, "import_graph": import_graph, "packages": packages, "architecture": architecture, "classifier": classifier};
}
export { buildDependencyGraph, buildImportGraph, buildTopology, classifyRepo, detectPackageManagers, extractApiSurface, inferArchitecture };
